bin/ruby-build

```diff
@@ -1119,18 +1133,45 @@ build_package_mac_openssl() {
   # gives precedence to the last -j option, so we can override that.
   package_option openssl make -j 1
 
-  # Use install_sw install_ssldirs instead of install to skip building docs which is slow.
-  # OpenSSL 1.1+ also needs install_ssldirs, 1.0 does not have that target.
-  if [[ "$1" == openssl-1.0.* ]]; then
-    MAKE_INSTALL_TARGET="install_sw" build_package_standard "$@"
+  if is_mac; then
+    # Use install_sw install_ssldirs instead of install to skip building docs which is slow.
+    # OpenSSL 1.1+ also needs install_ssldirs, 1.0 does not have that target.
+    if [[ "$1" == openssl-1.0.* ]]; then
+      MAKE_INSTALL_TARGET="install_sw" build_package_standard "$@"
+    else
+      MAKE_INSTALL_TARGET="install_sw install_ssldirs" build_package_standard "$@"
+    fi
   else
-    MAKE_INSTALL_TARGET="install_sw install_ssldirs" build_package_standard "$@"
+    build_package_standard "$@"
```

Comment:
We should also do the optimization to not build docs on non-macOS, so I think this section of the code shouldn't change.

Reply:
I misunderstand error: `ln: failed to create symbolic link '': No such file or directory`.
I thought it happened by `MAKE_INSTALL_TARGET`.
But it caused by bug mentioned at <https://github.com/rbenv/ruby-build/pull/1974#discussion_r874422761>.